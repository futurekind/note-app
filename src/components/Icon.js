import React, { PropTypes } from 'react';
import styled from 'styled-components';

const getIcon = ({ id }) => {
    switch(id) {
        case 'add':
            return <path fillRule="nonzero" d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6"/>;
        
        case 'check':
            return <path fillRule="nonzero" d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41"/>
        
        case 'down':
            return <path fillRule="nonzero" d="M7.41 8.84L12 13.42l4.59-4.58L18 10.25l-6 6-6-6"/>

        case 'edit':
            return <path d="M6 15.5V18h2.5l7.373-7.373-2.5-2.5L6 15.5zm11.807-6.807c.26-.26.26-.68 0-.94l-1.56-1.56c-.26-.26-.68-.26-.94 0l-1.22 1.22 2.5 2.5 1.22-1.22z" fillRule="nonzero"/>

        default:
            return null;
    }
}

const View = styled.svg`
    fill: currentColor;
`

const Icon = ({
    size,
    className,
    ...props
}) => {
    return (
        <View width={ size } height={ size } className={ className } viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
            { getIcon(props) }
        </View>
    )
}

Icon.defaultProps = {
    size: 24,
}

Icon.propTypes = {
    size: PropTypes.number,
    id: PropTypes.oneOf([
        'add', 'down', 'up', 'check', 'edit'
    ]),
}

export default Icon