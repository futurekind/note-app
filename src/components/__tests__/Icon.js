import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Icon from '../Icon';

describe('Icon', () => {
    const wrapper = shallow(
        <Icon />
    )

    it('it renders', () => {
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders width and height', () => {
        wrapper.setProps({
            size: 30
        })

        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders `add` icon', () => {
        wrapper.setProps({
            id: 'add'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders `check` icon', () => {
        wrapper.setProps({
            id: 'check'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders `down` icon', () => {
        wrapper.setProps({
            id: 'down'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders `up` icon', () => {
        wrapper.setProps({
            id: 'up'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders `edit` icon', () => {
        wrapper.setProps({
            id: 'edit'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

})