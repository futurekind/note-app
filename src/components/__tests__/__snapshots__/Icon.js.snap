exports[`Icon it renders 1`] = `
<styled.svg
  height={24}
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg" />
`;

exports[`Icon renders \`add\` icon 1`] = `
<styled.svg
  height={30}
  viewBox="0 0 24 24"
  width={30}
  xmlns="http://www.w3.org/2000/svg">
  <path
    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6"
    fillRule="nonzero" />
</styled.svg>
`;

exports[`Icon renders \`check\` icon 1`] = `
<styled.svg
  height={30}
  viewBox="0 0 24 24"
  width={30}
  xmlns="http://www.w3.org/2000/svg">
  <path
    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41"
    fillRule="nonzero" />
</styled.svg>
`;

exports[`Icon renders \`down\` icon 1`] = `
<styled.svg
  height={30}
  viewBox="0 0 24 24"
  width={30}
  xmlns="http://www.w3.org/2000/svg">
  <path
    d="M7.41 8.84L12 13.42l4.59-4.58L18 10.25l-6 6-6-6"
    fillRule="nonzero" />
</styled.svg>
`;

exports[`Icon renders \`edit\` icon 1`] = `
<styled.svg
  height={30}
  viewBox="0 0 24 24"
  width={30}
  xmlns="http://www.w3.org/2000/svg">
  <path
    d="M6 15.5V18h2.5l7.373-7.373-2.5-2.5L6 15.5zm11.807-6.807c.26-.26.26-.68 0-.94l-1.56-1.56c-.26-.26-.68-.26-.94 0l-1.22 1.22 2.5 2.5 1.22-1.22z"
    fillRule="nonzero" />
</styled.svg>
`;

exports[`Icon renders width and height 1`] = `
<styled.svg
  height={30}
  viewBox="0 0 24 24"
  width={30}
  xmlns="http://www.w3.org/2000/svg" />
`;
